src/portals/hope/features/claims/claim-details/ClaimSubclaims/hooks/use-binding.ts

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react'
+import {
+  ClaimTypeBinding,
+  useGetClaimTypeBindingsQuery,
+  useGetClaimTypesQuery,
+} from 'types/generated/graphql'
+
+export const useBinding = ({
+  bindingType,
+  changeBindingType,
+  selectedClaimType,
+  selectedContractType,
+}: {
+  bindingType?: ClaimTypeBinding | null
+  changeBindingType?: (type: ClaimTypeBinding | null) => void
+  selectedClaimType?: string | null
+  selectedContractType?: string | null
+}): {
+  availableTypes: string[]
+  getBindingType: (claimType: string) => ClaimTypeBinding | null
+} => {
+  const { data } = useGetClaimTypesQuery()
+  const { data: bindingData } = useGetClaimTypeBindingsQuery()
+
+  const claimTypes = data?.claimTypes ?? []
+  const bindingTypes = bindingData?.claimTypeBindings
+
+  const [availableTypes, setAvailableTypes] = useState<string[]>([])
+
+  const getBindingType = (claimType: string): ClaimTypeBinding | null =>
+    bindingTypes?.find(
+      (type) =>
+        type?.claimType === claimType &&
+        type?.insuranceType === selectedContractType,
+    ) || null
+
+  useEffect(() => {
+    if (
+      selectedClaimType &&
+      selectedClaimType !== bindingType?.claimType &&
+      changeBindingType
+    ) {
+      changeBindingType(getBindingType(selectedClaimType))
+    }
+  }, [selectedClaimType])
+
+  useEffect(() => {
+    if (claimTypes && selectedContractType && bindingTypes) {
+      const filteredTypes = claimTypes.filter(
+        (claimType) => !!getBindingType(claimType),
+      )
+
+      if (selectedClaimType && !filteredTypes.includes(selectedClaimType)) {
+        // Removing selected type
+      }
+
+      setAvailableTypes(filteredTypes)
+    }
+  }, [claimTypes, selectedContractType, bindingTypes])
```

Comment:
Can't we simply map out the `type` of the binding instead of fetching the `claimTypes` separately? 
```suggestion
  }, [claimTypes: typeBindings.map((binding) => binding.type), selectedContractType, bindingTypes])
```